midterm-2023-11-06/code/5115-nikola-tosic/script.ts

```diff
@@ -0,0 +1,162 @@
+interface Album {
+    name: string;
+    year: number;
+}
+
+interface Band {
+    id: number;
+    name: string;
+    formed: number;
+    location: string;
+    country: string;
+    members: string[];
+    albums: Album[];
+}
+
+type BandSorter = (first: Band, second: Band) => number;
+
+document.addEventListener("DOMContentLoaded", siteCode);
+
+let bands: Band[] = [];
+
+async function siteCode() {
+    const data = await loadData();
+    bands = data;
+
+    fillCountries(bands);
+
+    displayBands(bands);
+
+    const nameSort = document.getElementById("sort-name")!;
+    nameSort.addEventListener("click", sortByName);
+
+    const idSort = document.getElementById("sort-id")!;
+    idSort.addEventListener("click", sortById);
+
+    const applyFilterButton = document.getElementById("apply-filter")!;
+    applyFilterButton.addEventListener("click", applyFilter);
+
+    const modal = document.getElementById("album-details")!;
+    modal.addEventListener("click", () => {
+        modal.classList.add("hidden");
+    });
+}
+
+const nameSorter: BandSorter = (first, second) => first.name.localeCompare(second.name);
```

Comment:
The functions for sorting are implemented file, but it's impossible to actually use them from the HTML. +10
Extra +5 for using the `slice` method.